src/Libraries/Microsoft.Extensions.Http.AutoClient/README.md

```diff
@@ -1 +1,229 @@
+
+The HTTP Auto Client methods must always have a `CancellationToken` as a parameter. This parameter is used to cancel the request if needed.
+
+```csharp
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Http.AutoClient;
+
+[AutoClient("MyHttpClient")]
+public interface IMyDependencyClient
+{
+    [Get("/api/users")]
+    Task<List<User>> GetUsers(CancellationToken cancellationToken = default);
+}
+```
+
+### URL parameters
+
+The URL may contain parameters. For example `/api/users/{userId}`. You can define these as parameters of the Auto Client method. The name of the parameters in the method and URL must match.
+
+```csharp
+using System.Threading.Tasks;
+using Microsoft.Extensions.Http.AutoClient;
+
+[AutoClient("MyHttpClient")]
+public interface IMyDependencyClient
+{
+    [Get("/api/users/{userId}")]
+    public Task<User> GetUser(string userId, CancellationToken cancellationToken = default);
+}
+```
+
+### HTTP Payloads
+
+To send a payload with your request, you mus tuse the `BodyAttribute` on a method parameter. If you don't pass any parameter to it, it will consider the content type is JSON, serializing your parameter before sending. Otherwise, you can choose a `BodyContentType` and use it within the `BodyAttribute`.
+
+```csharp
+using System.Threading.Tasks;
+using Microsoft.Extensions.Http.AutoClient;
+
+[AutoClient("MyHttpClient")]
+public interface IMyDependencyClient
+{
+    [Post("/api/users")]
+    public Task<User> CreateUserAsync([Body] User user);
+
+    [Put("/api/users/{userId}/displayName")]
+    public Task<User> UpdateDisplayName(string userId, [Body(BodyContentType.TextPlain)] string displayName, CancellationToken cancellationToken = default);
+}
+```
+
+### Headers
+
+There are two ways of sending headers with your HTTP request. One of them is best suited for headers that never change value and are static. The other is for headers that change based on the parameters of your Auto Client method.
+
+#### Static Headers
+
+With the `StaticHeaderAttribute`, you can define a static header to be used with every request from the Auto Client. You must pass the header name and value to its constructor.
+
+```csharp
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Http.AutoClient;
+
+[AutoClient("MyHttpClient")]
+[StaticHeader("X-MyHeader", "HeaderValue")]
+public interface IMyDependencyClient
+{
+    [Get("/api/users")]
+    [StaticHeader("X-MethodHeader", "HeaderValue")]
+    public Task<List<User>> GetUsers(CancellationToken cancellationToken = default);
+}
+```
+
+#### Parameter Headers
+
+With the `HeaderAttribute`, you can receive the value for a header from the attributes of your Auto Client method. You must pass the header name to its constructor.
```

Comment:
I think it's worth mentioning that the value is not serialized! Any complex structures passed as an header value will just have its "ToString()" method called.